### PR TITLE
Open API Updates AAH-57

### PR DESCRIPTION
**Pull request by alikins**
WIP for getting a openapi spec we can use with c.r.c

Issue: AAH-57

### CODE REVIEW

File: galaxy_ng/app/settings.py
Comment (awcrosby) [inline galaxy_ng/app/settings.py:87]: are these lines something that can be removed since title is below?

File: galaxy_ng/app/settings.py
Comment (awcrosby) [inline galaxy_ng/app/settings.py:95]: should we link to galaxy_ng license?

File: galaxy_ng/app/common/openapi.py
Comment (awcrosby) [inline galaxy_ng/app/common/openapi.py:15]: curious what `preprocess_debug_logger()` is used for... and does `log.debug(...)` sometimes evaluate to a boolean false by not returning any result?

**Review by awcrosby — APPROVED** [review]
nothing blocking merging from my pov, but did add a couple clarifying questions.

File: galaxy_ng/app/settings.py
Comment (alikins) [reply galaxy_ng/app/settings.py:87]: Yeah, that is meant as an example of how to set drf_spectular settings vis env variables / dynaconf. I'll rephrase to make that clearer.

File: galaxy_ng/app/settings.py
Comment (alikins) [reply galaxy_ng/app/settings.py:95]: Ah, yeah. Good catch.

File: galaxy_ng/app/common/openapi.py
Comment (alikins) [reply galaxy_ng/app/common/openapi.py:15]: nah, log.debug() always returns nothing. That is leftover from some c&p from the other preprocess function

Comment (alikins) [reply galaxy_ng/app/common/openapi.py:15]: I take it back, forgot it was in the 'if' clause of the generator, and the if `not log.debug()` ends up if True, so all the elements are returned.

Comment (alikins) [reply galaxy_ng/app/common/openapi.py:15]: I've refactored that a bit to look less weird. I'll update after a './compose build' cycle to catch up some deps to test.

**Comment by hendersonreed** [comment]
A lot of the api's endpoints take specific parameters when POSTing- is there a way to generate request samples for those? Currently it looks like there's only request samples for the PATCH and PUT listings.

**Comment by alikins** [comment]
@hendersonreed Yeah, lots of stuff is incomplete at the moment. It can be corrected, either by tweaks to the spec generator, or direct markup on the viewsets and serializers.

**Comment by hendersonreed** [comment]
Seems to me that ideally we do those QoL improvements before merging this? I'm happy to contribute code if I can, since I'll also be a consumer of the final result.

**Pull request merged**
